backend/qemu.pm

```diff
@@ -405,33 +405,34 @@ sub start_qemu {
 
     mkpath($basedir);
 
-    if (!$vars->{KEEPHDDS} && !$vars->{SKIPTO}) {
-
-        # fresh HDDs
-        for my $i (1 .. $vars->{NUMDISKS}) {
-            no autodie qw(unlink);
-            unlink("$basedir/l$i");
-            if (-e "$basedir/$i.lvm") {
-                symlink("$i.lvm", "$basedir/l$i") or die "$!\n";
-                runcmd("/bin/dd", "if=/dev/zero", "count=1", "of=$basedir/l1");    # for LVM
-            }
-            elsif ($vars->{"HDD_$i"}) {
-                runcmd($qemuimg, "create", "$basedir/$i", "-f", "qcow2", "-b", $vars->{"HDD_$i"});
-                symlink($i, "$basedir/l$i") or die "$!\n";
-            }
-            else {
-                runcmd($qemuimg, "create", "$basedir/$i", "-f", $vars->{HDDFORMAT}, $vars->{HDDSIZEGB} . "G");
-                symlink($i, "$basedir/l$i") or die "$!\n";
-            }
+    my $keephdds = $vars->{KEEPHDDS} || $vars->{SKIPTO};
+
+    # fresh HDDs
+    for my $i (1 .. $vars->{NUMDISKS}) {
+        # skip HDD refresh when HDD exists and KEEPHDDS or SKIPTO is set
+        next if ($keephdds && (-e "$basedir/$i.lvm" || -e "$basedir/$i"));
+        no autodie qw(unlink);
+        unlink("$basedir/l$i");
+        if (-e "$basedir/$i.lvm") {
+            symlink("$i.lvm", "$basedir/l$i") or die "$!\n";
+            runcmd("/bin/dd", "if=/dev/zero", "count=1", "of=$basedir/l1");    # for LVM
```

Comment:
I'm not sure we even use LVM anywhere, so I just copied it without much thinking.